Feature/multiple templates arguments (#1096)
ViewComponent/view_component: docs/guide/templates.md

class MyLinkComponent < LinkComponent
end
```

> ## Multiple templates with arguments

Reviewer: ```suggestion
## Template parts
```

> ViewComponents can render multiple templates defined in the sidecar directory and send arguments to it:

Reviewer: ```suggestion
ViewComponents can render multiple templates defined in the sidecar directory:
```

```console
app/components
├── ...
├── test_component.rb
├── test_component
|   ├── list.html.erb
|   └── summary.html.erb
├── ...
```

> Templates are compiled to methods in the format `render_#{template_basename}(locals = {})`, which can then be called in the component.

Reviewer: ```suggestion
Template parts can then be rendered by calling `render_#{template_basename}`. Template parts accept a single positional argument that is accessible in the template as `locals`, defaulting to an empty hash if no argument is passed.
```
